### PR TITLE
ATDM Trilinos README.md documentation

### PR DESCRIPTION
**Pull request by bartlettroscoe**
@trilinos/<teamName>

## Description

This makes some improvements to the ATDM Trilinos README.md file.  See the commits for more details.

This also fixed a misspelling of 'tweaks' in the `ATDMDevEnvSettings.cmake` file.

## Motivation and Context

There was some confusion about the ATDM Trilinos configuration on some platforms and there were some typos and other mistakes.

## How Has This Been Tested?

To test the changes to the `ATDMDevEnvSettings.cmake` file, on 'ceerws1113` I ran:

```
$  ./checkin-test-atdm.sh gnu-debug --enable-packages=Teuchos --local-do-all
```

which returned:

```
PASSED (NOT READY TO PUSH): Trilinos: crf450.srn.sandia.gov

Wed Sep 26 17:32:05 MDT 2018

Enabled Packages: Teuchos

Build test results:
-------------------
0) MPI_RELEASE_DEBUG_SHARED_PT_OPENMP => Test case MPI_RELEASE_DEBUG_SHARED_PT_OPENMP was not run! => Does not affect push readiness! (-1.00 min)
1) gnu-debug => passed: passed=135,notpassed=0 (0.74 min)
```

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: cmake/std/atdm/README.md [hunk 1]
```diff
@@ -534,8 +558,8 @@ contents:
     environment.sh  # Load env for the given system based on $JOB_NAME keys
     all_supported_builds.sh  # [Optional] List of all supported builds
     tweaks/
-       <COMPILER0>-<BUILD_TYPE0>-<NODE_TYPE0>.cmake  # [Optional]
-       <COMPILER1>-<BUILD_TYPE1>-<NODE_TYPE1>.cmake  # [Optional]
+       <COMPILER0>-<BUILD_TYPE0>-<NODE_TYPE0>-<KOKKOS_ARCH0>.cmake  # [Optional]
+       <COMPILER1>-<BUILD_TYPE1>-<NODE_TYPE1>-<KOKKOS_ARCH0>.cmake  # [Optional]
```
Comment (mhoemmen) [inline cmake/std/atdm/README.md]: How come this isn't `KOKKOS_ARCH1`?

Comment (bartlettroscoe) [reply cmake/std/atdm/README.md]: I think this was a typo on my part but this is actually more accurate.  For example, these could be:

```
GNU-DEBUG-SERIAL-HSW.cmake
INTEL-RELEASE-OPENMP-HSW.cmake
```

Hopefully people get the idea anyway.

Comment (mhoemmen) [reply cmake/std/atdm/README.md]: Thanks Ross!

**Comment by bartlettroscoe** [comment]
@trilinos/framework,

Why is this PR not getting tested?  It was created 12 hours ago?

**Comment by jwillenbring** [comment]
@bartlettroscoe 

> Why is this PR not getting tested? It was created 12 hours ago?

There was another PR with a unicode character in a comment that was crashing the autotester. I attempted to "sanitize" that PR so this one can test. This happens from time to time and each unicode character is dealt with individually.

**Comment by bartlettroscoe** [comment]
> There was another PR with a unicode character in a comment that was crashing the autotester. 

@jwillenbring, can the autotester just be made robust to unicode?  We had a similar issue with some Pyhton code I worked with years ago and there are some python functions to do the conversion for you.

**Comment by jwillenbring** [comment]
> can the autotester just be made robust to unicode? We had a similar issue with some Pyhton code I worked with years ago and there are some python functions to do the conversion for you.

@bartlettroscoe I don't know the exact details of what @allevin does.

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe I don't know the exact details of what @allevin does.

@jwillenbring this is a bit of a serious problem.  How can figure where the unicode chars are?

**Pull request merged**